### DISCUSSION TITLE
question about injecting environment variable thorugh docker run --env

### DISCUSSION THREAD

**Discussion by hyun11732** [expatfile/next-runtime-env#85] [Q&A]
Hi, I am really suffering from trying to inject environment varaible through docker run --env and i found your package.

In Next JS, I tried

const temp - () = > {
const { NEXT_PUBLIC_FOO  } = useEnvContext();
console.log(NEXT_PUBLIC_FOO );
};
export default temp;

and did docker run XXX --env NEXT_PUBLIC_FOO=hi

but console.log still prints undefined.... I dont think it recognizes my runtime environment variable injected from docker run --env.

Is there any solution?

Thank you!! please help me !!!

**Comment by HofmannZ** [comment]
@hyun11732 - Did you add the `PublicEnvProvider` to your layout? And this you add `export const dynamic = 'force-dynamic';` to that layout?

See: https://github.com/expatfile/next-runtime-env#getting-started-

**Comment by tom-searle** [comment]
Hi @HofmannZ! With reference to [this thread](https://github.com/expatfile/next-runtime-env/issues/40) too, I am finding that I cannot use next-runtime-env with docker run and believe that it is due to using a multistage dockerfile (an equivalent single stage works just fine). Is there any reason that you are aware of here or special measures to take to allow it to work? Thank you
